src/Storages/Statistics/ConditionEstimator.cpp

```diff
@@ -87,7 +137,7 @@ std::pair<std::string, Float64> ConditionEstimator::extractBinaryOp(const RPNBui
     return std::make_pair(function_name, value);
 }
 
-Float64 ConditionEstimator::estimateSelectivity(const RPNBuilderTreeNode & node) const
+Float64 ConditionSelectivityEstimator::estimateRowCount(const RPNBuilderTreeNode & node) const
```

Comment:
Maybe in a follow-up PR: In l. 132, I think we can easily support other data types: Int32, UInt32, Int16, ...

Reply:
Field::Types only has very few types: https://github.com/ClickHouse/ClickHouse/blob/master/src/Core/Field.h#L322